osf/models/node.py

```diff
@@ -1191,6 +1191,10 @@ def add_contributor(self, contributor, permissions=None, visible=True,
         if contrib_to_add.is_disabled:
             raise ValidationValueError('Deactivated users cannot be added as contributors.')
 
+        if not contrib_to_add.is_registered and not contrib_to_add.unclaimed_records:
+            raise ValidationValueError('This contributor cannot be added. If the problem persists please report it '
```

Comment:
Drive-by: So is this a case that *should* not happen? If so, I'm not sure 400 is correct. 400 means that the client sent an invalid request. If we hit this case, it means that the server is in an invalid state, so a 500-series error would be more correct.

At the very least, we need to be notified about it, so an error should get logged to Sentry. Returning a 500-series response would do that automatically.